### DISCUSSION TITLE
How to set up a job to retrieve files from S3 with secrets, or any other authentication option?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17603] [Q&A] [answered]
regarding aws connection:
can someone point me to an example of how to setup a job to retrieve files from S3 with secrets,  or any other authentication option? I looked through but the documentation is kinda incomplete.

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1698851105485309?thread_ts=1698851105.485309&cid=C01U954MEER).*

**Comment by yuhan** [comment, marked as answer]
Here is an example:
```python
from dagster import asset
from dagster_aws.s3 import S3Resource

@asset
def my_asset(s3: S3Resource):
    ...


defs = Definitions(
    assets=[my_asset],
    resources={
        "s3": S3Resource(
            region_name="us-west-1",
            aws_access_key_id=EnvVar("AWS_ACCESS_KEY_ID"),
            aws_secret_access_key=EnvVar("AWS_SECRET_ACCESS_KEY"),
            aws_session_token=EnvVar("AWS_SESSION_TOKEN"),
        ),
    },
)
```

Check out https://docs.dagster.io/guides/dagster/using-environment-variables-and-secrets for our recommendations of handling env vars and secrets.
